### PR TITLE
deal with comments and #if's in argument lists

### LINKED ISSUE #410: When #if starts with comma, it formats oddly
In the given code, the `,` cannot be outside the `#if`. This leads to it being formatted like so.
Can we catch this and avoid breaking after the `,`?
```c#
            [Option(
                "server_port"
#if DEBUG
                ,
                Default = 50052
#endif
            )]
```

### PR DESCRIPTION
**Pull request by belav**
closes #410

### CODE REVIEW

File: Src/CSharpier/SyntaxPrinter/ArgumentListLikeSyntax.cs [hunk 1]
```diff
@@ -12,14 +12,17 @@ FormattingContext context
...
-                is ParenthesizedLambdaExpressionSyntax
-                    {
-                        ParameterList.Parameters.Count: 0,
-                        Block: { }
-                    }
-                    or SimpleLambdaExpressionSyntax { Block: { } }
+            arguments
+            is [
+                {
+                    Expression: ParenthesizedLambdaExpressionSyntax
```
Comment (shocklateboy92) [inline Src/CSharpier/SyntaxPrinter/ArgumentListLikeSyntax.cs:18]: oooh. Fancy list pattern match syntax :beach_umbrella:

**Review by shocklateboy92 — COMMENTED** [review]
TBH, I'm not sure #831 should be a bug.
It's clear in this case that there's a comma that's in the `#if`.
But it doesn't really belong to the next parameter, and we've never absorbed that line break in any other circumstance. 

IMHO I think the current behavior is better than putting the comma on the same line as the next argument. What do you think?

File: Src/CSharpier/SyntaxPrinter/ArgumentListLikeSyntax.cs (hunk 1 above)
Comment (belav) [reply Src/CSharpier/SyntaxPrinter/ArgumentListLikeSyntax.cs:18]: that was all rider...

**Comment by belav** [comment]
I think I'm okay with just not doing this. I ran into it a couple times at work with our new multitargeting but it is super rare. I think there were ~30 instances of it across all of the code in https://github.com/belav/csharpier-repos. It's kinda ugly as is, but also weird with the preceding comma. So let's just not add extra code to deal with the super rare thing when the new formatting is debatable.

**Pull request closed without merging**